File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let numSet = new Set(nums);
        let longest = 0;

        for(let num of nums){
            if(!numSet.has(num - 1)){
                let length = 0;
                while (numSet.has(length + num)){
                    length += 1;
                }
                longest = Math.max(longest,length);
            }
        }
        return longest;
    }
}

File: Data Structures & Algorithms/missing-number/submission-3.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        
        int n = nums.size() + 1;
        int xorAll = 0;
        int xorNum = 0;

        for(int i = 0 ; i < n; i++){
            xorAll = xorAll ^ i;
        }

        for(int num : nums){
            xorNum = xorNum ^ num;
        }

        return xorAll ^ xorNum;
    }
};
